packages/content-conformance/src/rules/no-unlinked-pages.js

```diff
@@ -0,0 +1,121 @@
+import flat from 'flat'
+
+const productSlugs = [
+  'packer',
+  'terraform',
+  'vault',
+  'boundary',
+  'consul',
+  'nomad',
+  'waypoint',
+  'vagrant',
+  'hcp',
+]
+
+let flattenedNavData = []
+
+function getFlattenedNavData(dataFiles) {
+  let result = []
+
+  if (flattenedNavData.length > 0) {
+    // only compute this once
+    return flattenedNavData
+  }
+
+  for (const file of dataFiles) {
+    // by convention, nav data files are those that end in nav-data.json
+    if (!file.basename.endsWith('nav-data.json')) {
+      continue
+    }
+
+    const navData = file.contents()
+
+    const basePath = file.basename.split('-nav-data.json')[0]
+    // ex. /cloud.docs.agents(\/|$)/i
+    const basePathPattern = new RegExp(
+      `^${basePath.replaceAll('-', '.')}(/|$)`,
+      'i'
+    )
+
+    // removes /{productSlug}/{basePath}/ from a nav path
+    const normalizeNavDataPathPattern = new RegExp(
+      `^(/(${productSlugs.join('|')})/)?${basePath.replace('-', '.')}`,
+      'i'
+    )
```

Comment:
We are close to a future where nav data will have developer-relative paths, as well as the current structure today. This normalizes the nav data paths to the current structure so we can consistently check both formats.

Reply:
Does this same logic need to be eventually applied to `ensure-valid-nav-paths`?

Reply:
Yeah, I would think so. Good call!